Implement derivatives for torch.remainder and torch.fmod wrt the second argument and update the docs

lezcano opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #67833
* #68188
* #67996
* #68523
* #68522
* #69909
* __->__ #69908

I also took this chance to clarify a bit the documentation of these
functions.

cc @brianjo @mruberry

Differential Revision: [D33774417](https://our.internmc.facebook.com/intern/diff/D33774417)

lezcano committed:
Implement derivatives for torch.remainder and torch.fmod wrt the second argument.

I also took this chance to clarify a bit the documentation of these
functions.

[ghstack-poisoned]

Lezcano commented:
Tagged @albanD and @nikitaved for the derivative and @mruberry for the docs.

lezcano committed:
Update on "Implement derivatives for torch.remainder and torch.fmod wrt the second argument and update the docs"


I also took this chance to clarify a bit the documentation of these
functions.

cc brianjo mruberry

[ghstack-poisoned]

albanD commented on torch/_torch_docs.py:
```diff
@@ -8287,10 +8292,16 @@ def merge_dicts(*dicts):
-Unlike :func:`torch.fmod`, however, if the sign of the modulus is different
-than the sign of the divisor :attr:`other` then the divisor is added to the modulus.
+Computes
+`Python's modulus operation <https://docs.python.org/3/reference/expressions.html#binary-arithmetic-operations>`_
```
Why the newline?

Lezcano replied:
The linter was complaining, and I did not want to figure out what's the correct way to split that hyperlink in sphinx :P

albanD replied:
ok!

albanD reviewed (commented):
Sounds good.

nikitaved reviewed (approved):
Forward/Backward AD LGTM!

albanD reviewed (approved):
Looks good on my end.
Leaving it to @mruberry for the docs review and landing.

lezcano committed:
Update on "Implement derivatives for torch.remainder and torch.fmod wrt the second argument and update the docs"


I also took this chance to clarify a bit the documentation of these
functions.

cc brianjo mruberry

[ghstack-poisoned]

lezcano committed:
Update on "Implement derivatives for torch.remainder and torch.fmod wrt the second argument and update the docs"


I also took this chance to clarify a bit the documentation of these
functions.

cc brianjo mruberry

[ghstack-poisoned]

lezcano committed:
Update on "Implement derivatives for torch.remainder and torch.fmod wrt the second argument and update the docs"


I also took this chance to clarify a bit the documentation of these
functions.

cc brianjo mruberry

[ghstack-poisoned]

lezcano committed:
Update on "Implement derivatives for torch.remainder and torch.fmod wrt the second argument and update the docs"


I also took this chance to clarify a bit the documentation of these
functions.

cc brianjo mruberry

[ghstack-poisoned]

lezcano committed:
Update on "Implement derivatives for torch.remainder and torch.fmod wrt the second argument and update the docs"


I also took this chance to clarify a bit the documentation of these
functions.

cc brianjo mruberry

[ghstack-poisoned]

lezcano committed:
Update on "Implement derivatives for torch.remainder and torch.fmod wrt the second argument and update the docs"


I also took this chance to clarify a bit the documentation of these
functions.

cc brianjo mruberry

[ghstack-poisoned]

lezcano committed:
Update on "Implement derivatives for torch.remainder and torch.fmod wrt the second argument and update the docs"


I also took this chance to clarify a bit the documentation of these
functions.

cc brianjo mruberry

[ghstack-poisoned]

lezcano committed:
Update on "Implement derivatives for torch.remainder and torch.fmod wrt the second argument and update the docs"


I also took this chance to clarify a bit the documentation of these
functions.

cc brianjo mruberry

[ghstack-poisoned]

lezcano committed:
Update on "Implement derivatives for torch.remainder and torch.fmod wrt the second argument and update the docs"


I also took this chance to clarify a bit the documentation of these
functions.

cc brianjo mruberry

[ghstack-poisoned]

lezcano committed:
Update on "Implement derivatives for torch.remainder and torch.fmod wrt the second argument and update the docs"


I also took this chance to clarify a bit the documentation of these
functions.

cc brianjo mruberry

[ghstack-poisoned]

lezcano committed:
Update on "Implement derivatives for torch.remainder and torch.fmod wrt the second argument and update the docs"


I also took this chance to clarify a bit the documentation of these
functions.

cc brianjo mruberry

[ghstack-poisoned]

lezcano committed:
Update on "Implement derivatives for torch.remainder and torch.fmod wrt the second argument and update the docs"


I also took this chance to clarify a bit the documentation of these
functions.

cc brianjo mruberry

[ghstack-poisoned]

soulitzer commented:
Updated https://github.com/pytorch/pytorch/issues/71117 to link to this.

lezcano committed:
Update on "Implement derivatives for torch.remainder and torch.fmod wrt the second argument and update the docs"


I also took this chance to clarify a bit the documentation of these
functions.

cc brianjo mruberry

[ghstack-poisoned]

lezcano committed:
Update on "Implement derivatives for torch.remainder and torch.fmod wrt the second argument and update the docs"


I also took this chance to clarify a bit the documentation of these
functions.

cc brianjo mruberry

[ghstack-poisoned]

lezcano committed:
Update on "Implement derivatives for torch.remainder and torch.fmod wrt the second argument and update the docs"


I also took this chance to clarify a bit the documentation of these
functions.

cc brianjo mruberry

[ghstack-poisoned]

mruberry commented:
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33774417).

lezcano committed:
Update on "Implement derivatives for torch.remainder and torch.fmod wrt the second argument and update the docs"


I also took this chance to clarify a bit the documentation of these
functions.

cc brianjo mruberry

Differential Revision: [D33774417](https://our.internmc.facebook.com/intern/diff/D33774417)

[ghstack-poisoned]

mruberry commented:
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33774417).

lezcano committed:
Update on "Implement derivatives for torch.remainder and torch.fmod wrt the second argument and update the docs"


I also took this chance to clarify a bit the documentation of these
functions.

cc brianjo mruberry

Differential Revision: [D33774417](https://our.internmc.facebook.com/intern/diff/D33774417)

[ghstack-poisoned]

mruberry commented:
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33774417).
